scala-vector-tile/src/main/scala/geotrellis/vectortile/Commands.scala

```diff
@@ -0,0 +1,129 @@
+                    Line(point_lists.head)
+                } else {
+                    MultiLine(point_lists.map(
+                        (ln: List[(Double, Double)]) => Line(ln)))
+                }
+
+            case POLYGON =>
+                if (point_lists.length == 1) {
+                    Polygon(point_lists.head)
+                } else {
+                    MultiPolygon(point_lists.map(
+                        (pg: List[(Double, Double)]) => Polygon(pg)))
+                }
+
+            case _ =>
+                // this should probably be logged.
+                null
+
+        }
+
+
+        def interpret_commands() {
+
+            val point_list: ListBuffer[(Double, Double)] =
+                ListBuffer.empty[(Double, Double)]
+            var (x: Int, y: Int) = (0, 0)
+            var idx: Int = 0
+
+            def zigZagDecode(n: Int) = ((n >> 1) ^ (-(n & 1)))
+
+            while(idx < commands.length) {
+
+                var command = commands(idx)
+                var (id, count) = (command & 0x7, command >> 3)
+                idx += 1
+
+                for(_ <- 0 until count) {
+
+                    id match {
+
+                        case MoveTo =>
+                            if (!point_list.isEmpty) {
+                                point_lists += point_list.toList
+                                point_list.clear
+                            }
+                            x += zigZagDecode(commands(idx))
+                            y += zigZagDecode(commands(idx+1))
+                            idx += 2
+                            point_list += ((x / scale, y / scale))
+
+                        case LineTo =>
+                            if (point_list.isEmpty) {
+                                // this is unexpected and should be logged
+                            }
+                            val dx = zigZagDecode(commands(idx))
+                            val dy = zigZagDecode(commands(idx+1))
+                            x += dx
+                            y += dy
+                            idx += 2
+                            if (dx != 0 || dy != 0)
+                                point_list += ((x / scale, y / scale))
+
+                        case ClosePath =>
+                            if (point_list.isEmpty) {
+                                    // this is unexpected and should be logged
+                                } else {
+                                    if (point_list.head != point_list.last) {
+                                        point_list += point_list.head
+                                    }
+                                    point_lists += point_list.toList
+                                    point_list.clear
+                                }
+
+                    }
+
+                }
+
+            }
+
+            if (!point_list.isEmpty) { point_lists += point_list.toList }
+
+        }
+
+        return null
```

Comment:
`return` statement unnecessary. Also, never return nulls (this is something that Java does that Scala is very against doing)